framework/auth/views.py

```diff
@@ -258,6 +258,11 @@ def login_and_register_handler(auth, login=True, campaign=None, next_url=None, l
         if service_url:
             data['next_url'] = service_url
 
+    # special case for preprints campaign url
+    if not logout and not campaign and not login and not auth.logged_in:
```

Comment:
handles the the case when logged out user hits the new preprints campaign url: `https://osf.io/login?next=https://osf.io/preprints/`.